cli/asc.js

```diff
@@ -342,9 +370,72 @@ exports.main = function main(argv, options, callback) {
           }
         }
       }
+      /*
+      In this case the library wasn't found so we check paths
+      */
+      if (sourceText == null) {
+        if (args.traceResolution) {
+            stderr.write("Looking for " + sourcePath + " imported by " + dependee + " " + EOL);
+        }
+        paths = getPaths(path.join(baseDir, dependee));
+        let _package = sourcePath.replace(/\~lib\/([^\/]*).*/, "$1");
+        for (let _path of paths) {
+          if (args.traceResolution) {
+            stderr.write("    in " + _path);
+          }
+          let ascMain = (() => {
+            if (packages.has(_package)) {
+              return packages.get(_package);
+            }
+            let p = path.join(_path, _package, "package.json");
+            let res = readFile(p, baseDir);
+            if (res) {
+              let package_json;
+              try {
+                package_json = JSON.parse(res);
+              } catch (e) {
+                return callback(Error("Parsing " + p + " failed"));
+              }
+              let mainFile = package_json.ascMain;
+              if (mainFile) {
+                let newPackage = mainFile.replace(/(.*)\/index\.ts/, '$1');
+                packages.set(_package, newPackage);
+                return newPackage;
+              }
+            }
+            return "assembly";
+          })()
+          let realPath = (_p) => {
+            return _p.replace(/\~lib\/([^/]*)\/(.*)/, path.join(_path, "$1", ascMain, "$2"));
```

Comment:
Seems that this breaks if a directory or file contains `$1` or `$2`.

Reply:
You mean `"/~lib/$1/$2.ts"`? I think it should be ok

Reply:
I meant that, if `_path` or `ascMain` contains the substring `$1` or `$2` already, the substitution would do the wrong thing. Not sure how it's on Linux, but on Windows a file name like `file$1.ts` is valid.

Reply:
right